harvester/harvest/alma.py

```diff
@@ -78,51 +89,94 @@ def parse_marc_records_from_files(self) -> Iterator[MARCRecord]:
                     while element.getprevious() is not None:
                         del element.getparent()[0]
 
-    def _list_xml_files(self) -> Iterator[str]:
-        """Provide list of MARC record set XML files filtered by harvest type and date.
+    def _get_latest_full_run_date(self, filepaths: list[str]) -> str:
+        """Get the date from the latest Alma full extract."""
+        filtered_filepaths = self._filter_filepaths_by_harvest_type(filepaths)
+        extracted_dates = sorted(
+            {
+                self._get_date_from_filepath(filepath)
+                for filepath in filtered_filepaths
+                if filepath
+            },
+            reverse=True,
+        )
+        if not extracted_dates:
+            raise AlmaCannotIdentifyLatestFullRunDateError
+        return extracted_dates[0]
 
-        Retrieve list of XML files from S3 or local filesystem, then yield filepaths that
-        match harvest options.
-        """
-        if self.input_files.startswith("s3://"):
-            filepaths = self._list_s3_xml_files()
-        else:
-            filepaths = self._list_local_xml_files()  # pragma: nocover
+    def _get_date_from_filepath(self, filepath: str) -> str:
+        """Get date string from filepath."""
+        match = FILEPATH_DATE_REGEX.match(filepath)
+        if not match:  # pragma: nocover
+            message = f"Could not parse date from filepath: {filepath}"
+            logger.warning(message)
+            return ""
```

Comment:
As mentioned in a comment below, if we want to handle the edge/error case when a date is not extracted, we might want this to return `None` instead of empty string.  Generally speaking, empty strings make me nervous.  Definitely have their place sometimes, but worth considering other options I think.

Reply:
I had the function return an empty string so that I could set the return type of `_get_date_from_filepath(...) -> str`, which allows us to avoid the Mypy error: 

```
harvester/harvest/alma.py:96: error: Value of type variable "SupportsRichComparisonT" of "sorted" cannot be "str | None"  [type-var]
```

Reply:
Please see the latest commit for the latest typehinting updates.

Reply:
Oops!  Maybe it was this comment that _my_ comment https://github.com/MITLibraries/geo-harvester/commit/898e3b22ccb542df3eb274579267a1068cbddebd#r1563099498 most directly relates to.